File: common/src/test/java/guepardoapps/kucahome/common/converter/JsonDataToWirelessSocketUnitTest.java
package guepardoapps.kucahome.common.converter;

import org.junit.Test;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.UUID;

import guepardoapps.lucahome.common.converter.wirelesssocket.JsonDataToWirelessSocketConverter;
import guepardoapps.lucahome.common.enums.common.ServerDatabaseAction;
import guepardoapps.lucahome.common.models.wirelesssocket.WirelessSocket;

import static org.junit.Assert.assertEquals;

public class JsonDataToWirelessSocketUnitTest {
    @Test
    public void conversion_isCorrect() throws Exception {
        // Arrange
        JsonDataToWirelessSocketConverter converter = new JsonDataToWirelessSocketConverter();

        String jsonResponse = "{\"Data\":[{\"WirelessSocket\":{\"Uuid\":\"UUID\",\"RoomUuid\":\"RoomUUID\",\"Name\":\"Name\",\"Code\":\"Code\",\"State\":1,\"LastTrigger\":{\"UserName\":\"LastTriggerUser\",\"Year\":2018,\"Month\":1,\"Day\":1,\"Hour\":6,\"Minute\":25}}}]}";

        WirelessSocket wirelessSocket = new WirelessSocket();
        wirelessSocket.uuid = UUID.fromString("UUID");
        wirelessSocket.roomUuid = UUID.fromString("RoomUUID");
        wirelessSocket.name = "Name";
        wirelessSocket.code = "Code";
        wirelessSocket.setState(true);
        wirelessSocket.lastTriggerDateTime = Calendar.getInstance();
        wirelessSocket.lastTriggerUser = "LastTriggerUser";
        wirelessSocket.setOnServer(true);
        wirelessSocket.setServerDatabaseAction(ServerDatabaseAction.Null);
        wirelessSocket.setChangeCount(0);
        wirelessSocket.setShowInNotification(true);

        ArrayList<WirelessSocket> expected = new ArrayList<>();
        expected.add(wirelessSocket);

        // Act
        ArrayList<WirelessSocket> actual = converter.parse(jsonResponse);

        // Assert
        assertEquals(expected.size(), actual.size());
        assertEquals(expected.get(0).getUuid(), actual.get(0).getUuid());
        assertEquals(expected.get(0).getRoomUuid(), actual.get(0).getRoomUuid());
        assertEquals(expected.get(0).getName(), actual.get(0).getName());
        assertEquals(expected.get(0).getCode(), actual.get(0).getCode());
        assertEquals(expected.get(0).getState(), actual.get(0).getState());
    }
}

File: common/src/test/java/guepardoapps/lucahome/common/converter/JsonDataToWirelessSocketUnitTest.kt
package guepardoapps.lucahome.common.converter

import org.junit.Test

import java.util.ArrayList
import java.util.Calendar
import java.util.UUID

import guepardoapps.lucahome.common.converter.wirelesssocket.JsonDataToWirelessSocketConverter
import guepardoapps.lucahome.common.enums.common.ServerDatabaseAction
import guepardoapps.lucahome.common.models.wirelesssocket.WirelessSocket

import org.junit.Assert.assertEquals

class JsonDataToWirelessSocketUnitTest {
    @Test
    @Throws(Exception::class)
    fun conversion_isCorrect() {
        // Arrange
        val converter = JsonDataToWirelessSocketConverter()

        val jsonResponse =
                "{" +
                        "\"Data\":" +
                        "[" +
                        "{\"WirelessSocket\":" +
                        "{" +
                        "\"Uuid\":\"c111d375-e729-4cf1-95c8-e5a0253586e1\"," +
                        "\"RoomUuid\":\"6a7acd45-f863-40ed-a02e-d2ef04609f00\"," +
                        "\"Name\":\"Name\"," +
                        "\"Code\":\"Code\"," +
                        "\"State\":1," +
                        "\"LastTrigger\":" +
                        "{" +
                        "\"UserName\":\"LastTriggerUser\"," +
                        "\"Year\":2018," +
                        "\"Month\":1," +
                        "\"Day\":1," +
                        "\"Hour\":6," +
                        "\"Minute\":25" +
                        "}" +
                        "}" +
                        "}" +
                        "]" +
                        "}"

        val wirelessSocket = WirelessSocket()
        wirelessSocket.uuid = UUID.fromString("c111d375-e729-4cf1-95c8-e5a0253586e1")
        wirelessSocket.roomUuid = UUID.fromString("6a7acd45-f863-40ed-a02e-d2ef04609f00")
        wirelessSocket.name = "Name"
        wirelessSocket.code = "Code"
        wirelessSocket.state = true
        wirelessSocket.lastTriggerDateTime = Calendar.getInstance()
        wirelessSocket.lastTriggerUser = "LastTriggerUser"
        wirelessSocket.isOnServer = true
        wirelessSocket.serverDatabaseAction = ServerDatabaseAction.Null
        wirelessSocket.changeCount = 0
        wirelessSocket.showInNotification = true

        val expected = ArrayList<WirelessSocket>()
        expected.add(wirelessSocket)

        // Act
        val actual = converter.parse(jsonResponse)

        // Assert
        assertEquals(expected.size.toLong(), actual.size.toLong())
        assertEquals(expected[0].uuid, actual[0].uuid)
        assertEquals(expected[0].roomUuid, actual[0].roomUuid)
        assertEquals(expected[0].name, actual[0].name)
        assertEquals(expected[0].code, actual[0].code)
        assertEquals(expected[0].state, actual[0].state)
    }
}